File: utils/connectors/src/org/monet/bpi/BPINodeReference.java
package org.monet.bpi;

public interface BPINodeReference {

  public String getLabel();

  public void setLabel(String label);

  public String getDescription();

  public void setDescription(String description);

  public String getAttributeValue(String code);
  
  public void setAttributeValue(String code, String value);
 
}

File: applications/space/client/office/src/core/model/bpinodereference.js
function CGBPINodeReference(NodeReference) {
  this.NodeReference = NodeReference;
};

//---------------------------------------------------------------------
CGBPINodeReference.prototype.getLabel = function () {
  return this.NodeReference.getLabel();
};

//---------------------------------------------------------------------
CGBPINodeReference.prototype.setLabel = function (sValue) {
  return this.NodeReference.setLabel(sValue);
};

//---------------------------------------------------------------------
CGBPINodeReference.prototype.getDescription = function () {
  return this.NodeReference.getDescription();
};

//---------------------------------------------------------------------
CGBPINodeReference.prototype.setDescription = function (sValue) {
  return this.NodeReference.setDescription(sValue);
};

//---------------------------------------------------------------------
CGBPINodeReference.prototype.getAttributeValue = function (code) {
  return this.NodeReference.getAttributeValue(code);
};

//---------------------------------------------------------------------
CGBPINodeReference.prototype.setAttributeValue = function (code, sValue) {
  return this.NodeReference.setAttributeValue(code, sValue);
};